arbor/communication/communicator.hpp

```diff
@@ -46,10 +46,10 @@ class communicator {
                           const domain_decomposition& dom_dec,
                           const execution_context* ctx)
     {
-        distributed_ = &ctx->distributed;
+        distributed_ = ctx->distributed;
         thread_pool_ = ctx->thread_pool;
 
-        num_domains_ = distributed_->size();
+        num_domains_ = distributed_.get()->size();
```

Comment:
`std::shared_ptr` overloads `operator->`, so you can replace `distributed_.get()->size()` with `distributed_->size()`.